infra/sec/application-gateway/application-gateway.bicep

```diff
@@ -0,0 +1,204 @@
+@description('Required. Name of the Application gateway.')
+param name string
+
+@description('Required. Environment name.')
+param environment string
+
+@description('Optional. The Azure region where the resources will be deployed.')
+param location string = resourceGroup().location
+
+@description('Required. The name of the SKU for the Application Gateway.')
+@allowed([
+  'WAF_v2'
+])
+param sku string = 'WAF_v2'
+
+
+@description('Required. The parameter object for the virtual network. The object must contain the name,skuName,resourceGroup and subnetPrivateEndpoints values.')
+param vnet object
+
+@description('Required. Environment name.')
+param publicIPName string
+
+@description('Required. The name of the Front Door WAF Policy to create.')
+param wafPolicyName string
+
+@description('Optional. The list of managed rule sets to configure on the WAF (DRS).')
+param managedRuleSets array = []
+
+@description('Optional. The PolicySettings object (state,mode) for policy.')
+param policySettings object = {
+  state: 'Enabled'
+  mode: 'Prevention'
+}
+
+@description('Required. The FrontDoor ID.')
+param frontDoorId string
+
+@description('Required. backends Object(backendAddressPool, backendHttpSetting, httpListener, requestRoutingRule)')
+param backends array
+
+@description('Optional. Date in the format yyyyMMdd-HHmmss.')
+param deploymentDate string = utcNow('yyyyMMdd-HHmmss')
+
+@description('Optional. Date in the format yyyy-MM-dd.')
+param createdDate string = utcNow('yyyy-MM-dd')
+
+var customTags = {
+  Location: location
+  CreatedDate: createdDate
+  Environment: environment
+}
+var tags = union(loadJsonContent('../../common/default-tags.json'), customTags)
+
+var applicationGatewayTags = {
+  Name: name
+  Purpose: 'ADP Application Gateway'
+  Tier: 'Shared'
+}
+
+var applicationGatewayID = '${resourceGroup().id}/providers/Microsoft.Network/applicationGateways/${name}'
+
+module appGWpublicIpAddress '.bicep/public-ip-address.bicep' = {
+  name: 'appGWpublicIpAddress-${deploymentDate}'
+  params: {
+    name: publicIPName
+    location: location
+    tags: tags
+  }
+}
+
+module applicationGatewayWAFPolicy '.bicep/application-gateway-waf-custom.bicep' = {
```

Comment:
is the WAF going in a separate resource group?
if not then why do we need a module for this?

Reply:
same for the public ip.

Reply:
My understanding is if we have module available in our bicep ACR then we can use it, Hence here i have simply used it for that purpose.

Reply:
Yes, we use the modules from the registry but why do you need a submodule here? can the waf policy and IP creation not be part of the application-gateway.bicep?

Reply:
Yes It can be a part of same template. 
But as these are different resources i wanted to keep it separate for maintenance and readability perspective hence I moved it to separate folder. let me know if thats fine?

Reply:
@ajaysoni-cap : Updated.